internal/provider/data_source_http.go

```diff
@@ -111,6 +111,11 @@ a 5xx-range (except 501) status code is received. For further details see
 				},
 			},
 
+			"allow_host_override": schema.BoolAttribute{
+				Description: "Allows the `Host` header defined in `request_headers` to override the host of the request. Defaults to `false`",
+				Optional:    true,
+			},
+
```

Comment:
I like the idea here, since `Host` is typically populated automatically, we'd avoid configurations accidentally overriding it (whereas the other headers in this field typically don't exist, so they're being set on the request). An added bonus of not breaking existing configurations that rely on this behavior today 😆 

If there will need to be explicit opt-in, what do you think about just moving the `Host` header value into this new attribute? Then reading the configuration you know, "here are the headers set on the request, and I'm overwriting the host because of my load balancer"

```terraform
data "http" "test" {
  url      = "https://www.example.com"
  host_override = "www.differentexample.com"
  request_headers = {
    # headers to set
  }
}
```

Reply:
Thinking back on this, I wonder if we can convince ourselves that we don't need an override flag/value at all? Tools like curl allow you to override the host header with the same mechanism as setting other headers, so maybe it's not that unexpected of a behavior for `request_headers` to mean "write or overwrite".

The risk would be existing `http` data sources that are setting a `Host` header in `request_headers`, but are expecting it to be ignored. Seems like an unlikely case at first thought, but of course -> https://www.hyrumslaw.com/

Reply:
Following an out of band conversation, we decided to remove the feature flag and treat the `request_header` override as a bug fix rather than a feature.